Bug: fix broken GitHub actions for test and build

### Describe the bug

The GitHub test and build actions are failing related to go dependencies:

```
Error: main.go:7:2: cannot find package "github.com/joho/godotenv" in any of:
	/opt/hostedtoolcache/go/1.2.2/x64/src/pkg/github.com/joho/godotenv (from $GOROOT)
	($GOPATH not set)
```

These should ideally run both
- `go test` on all packages in the repository.
- `go build` to ensure building the binary works.

### Steps to reproduce

https://github.com/open-sauced/pizza/actions

### Browsers

_No response_

### Additional context (Is this in dev or production?)

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

--- a/.github/workflows/test.yaml
+++ b/.github/workflows/test.yaml
@@ -1,17 +1,52 @@
-name: Test and build
+name: Lint, test, and build
 
-on: [push]
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types:
+      - opened
+      - edited
+      - synchronize
+      - reopened
+
+permissions:
+  # So golangci-lint can read the contents of the lint yaml file
+  contents: read
 
 jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Go
+        uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # v4.0.1
+        with:
+          go-version: 1.20.x
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: v1.53
+
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Go
+        uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # v4.0.1
+        with:
+          go-version: 1.20.x
+      - name: Test
+        run: make test
+
   build:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
       - name: Set up Go
-        uses: actions/setup-go@v3
+        uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # v4.0.1
         with:
-          go-version: 1.20
-      - name: Build
-        run: go build -v ./...
-      - name: Test
-        run: go test -v ./...
+          go-version: 1.20.x
+      - name: Build go binary
+        run: make build

--- a/.gitignore
+++ b/.gitignore
@@ -1,0 +1,5 @@
+# This is where the "make local" target drops build artifacts
+build/
+
+# explicitly ignore env files to ensure secrets aren't being accidentally committed
+.env

--- a/Makefile
+++ b/Makefile
@@ -1,3 +1,5 @@
+.PHONY: all lint test run local build setup-test-env teardown-test-env
+
 ROOT_DIR:=$(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
 
 all: lint test build
@@ -16,6 +18,9 @@ test:
 
 run:
 	go run main.go
+
+local:
+	go build -o build/pizza-oven main.go
 
 build:
 	docker build . -t pizza-oven:latest

